packages/blockchain-wallet-v4/src/types/KVStoreEntry.js

```diff
@@ -83,16 +81,16 @@ export const deriveMetadataNode = masterHDNode => {
   return masterHDNode.deriveHardened(purpose)
 }
 
-export const fromMetadataXpriv = curry((xpriv, typeId, network) =>
-  fromMetadataHDNode(Bitcoin.HDNode.fromBase58(xpriv, network), typeId)
-)
+export const fromMetadataXpriv = curry((xpriv, typeId, network) => {
+  return fromMetadataHDNode(Bitcoin.bip32.fromBase58(xpriv, network), typeId)
+})
 
 export const fromMetadataHDNode = curry((metadataHDNode, typeId) => {
   let payloadTypeNode = metadataHDNode.deriveHardened(typeId)
-  let node = payloadTypeNode.deriveHardened(0)
-  let privateKeyBuffer = payloadTypeNode.deriveHardened(1).keyPair.d.toBuffer()
+  let privateKeyBuffer = payloadTypeNode.deriveHardened(1).__D
```

Comment:
This does not look equivalent to what happened in the old one.

Reply:
Good call, I am not passing through the derived node at index 0 into the KV store.

Is there a reason why we are deriving another node for the encryption key at index 1 instead of just reusing the 0 indexed node?

Reply:
I'm sure there was some reason, but I can't remember. Maybe @xenog does?